src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -1778,7 +1758,8 @@ MulticopterPositionControl::control_position(float dt)
 			thrust_sp = math::Vector<3>(_pos_sp_triplet.current.a_x, _pos_sp_triplet.current.a_y, _pos_sp_triplet.current.a_z);
 
 		} else {
-			thrust_sp = vel_err.emult(_params.vel_p) + _vel_err_d.emult(_params.vel_d) + _thrust_int;
+			thrust_sp = vel_err.emult(_params.vel_p) + _vel_err_d.emult(_params.vel_d)
+				    + _thrust_int - math::Vector<3>(0, 0, _params.thr_hover);
 		}
```

Comment:
Makes no difference for the compiled output but for correctness you are totally right.